doc/Mutable_References.md

```diff
@@ -330,6 +331,20 @@ is effectively a wrapper for a `Cell` that contains an `Option`, like:
 `None` if the value is effectively unset. Using an `OptionalCell` (like a
 `NumCell`) makes the code clearer and hides extra tedious function calls.
 
+### Comparison to `TakeCell`
+
+`TakeCell` and `OptionalCell` are quite similar, but the key differentiator is
+the `Copy` bound required for items to use some of the methods defined on `OptionalCell`, such as `map()`.
+The `Copy` bound enables safe "reentrant" access to the stored value, because multiple accesses will be operating
+on a copy of the stored item. The semantic difference
```

Comment:
```suggestion
on different copies of the same stored item. The semantic difference
```
is that valid? If so, might be slightly more clear.

Reply:
yes I think that is correct